### PR TITLE
upgrade monster_property

### PR DESCRIPTION
**Pull request by JaskRendix**
after #1732 I was going to create a condition to get the shape (eg if there was an aquatic monster in the party), but I bumped into **monster_property**.

With my surprise **monster_property** didn't pop up anywhere, but I understood as soon as I looked inside. Slot conditions are tough, because at the end who cares which monster the player has in the first position, or in the second, how you can forecast... ?
I upgraded the condition by putting a loop for all the monsters in the party and new valid values. 

Now the condition is more powerful, but I realized that it can replace the existing **has_tech** (still not used in all the maps) and **has_monster** used 19 times (easily replaceable, **spyder_flower_shop** 7 and **spyder_flower_city** 12).

I'm referring to event conditions.

Let me know if you want to get rid of these two.

### CODE REVIEW

File: tuxemon/event/conditions/monster_property.py [hunk 1]
```diff
@@ -43,25 +48,46 @@ def test(self, session: Session, condition: MapCondition) -> bool:
...
+            elif prop == "stage":
+                if mon.stage == val:
+                    return True
+            # monster shape (eg. aquatic, brute, etc.)
+            elif prop == "shape":
+                if mon.shape == val:
+                    return True
+            # monster taste_cold (eg. mild, etc.)
+            elif prop == "taste_cold":
+                if mon.shape == val:
```
Comment (kerizane) [inline tuxemon/event/conditions/monster_property.py]: This

File: tuxemon/event/conditions/monster_property.py [hunk 2]
```diff
@@ -43,25 +48,46 @@ def test(self, session: Session, condition: MapCondition) -> bool:
...
+                    return True
+            # monster taste_warm (eg peppy, etc.)
+            elif prop == "taste_warm":
+                if mon.shape == val:
```
Comment (kerizane) [inline tuxemon/event/conditions/monster_property.py]: This

File: tuxemon/event/conditions/monster_property.py [hunk 3]
```diff
@@ -9,31 +9,36 @@
...
-            is monster_property <slot>,<property>,<value>
+            is monster_property <property>,<value>
 
     Script parameters:
-        slot: Position of the monster in the player monster list.
-        property: Property of the monster to check (e.g. "level"). Valid values
-            are:
-                - name
-                - level
-                - level_reached
```
Comment (kerizane) [inline tuxemon/event/conditions/monster_property.py:25]: Why did you remove level_reached?

Comment (JaskRendix) [reply tuxemon/event/conditions/monster_property.py:25]: I thought it was too similar to level, but I have added it again.

File: tuxemon/event/conditions/monster_property.py (hunk 1 above)
Comment (JaskRendix) [reply tuxemon/event/conditions/monster_property.py]: Thanks

File: tuxemon/event/conditions/monster_property.py (hunk 2 above)
Comment (JaskRendix) [reply tuxemon/event/conditions/monster_property.py]: Thanks

**Review by kerizane — APPROVED** [review]
